cluster/cluster.go

```diff
@@ -79,20 +79,31 @@ func Start() {
 	log.Info("CLU Start: joined to %d nodes in cluster", n)
 }
 
+type partitionCandidates struct {
+	priority int
+	nodes    []Node
+}
+
 // return the list of nodes to broadcast requests to
-// Only 1 member per partition is returned. This list includes
-// ThisNode if it is capable of handling queries.
+// Only 1 member per partition is returned.
+// The nodes are selected based on priority, using a random selection
+// if there are multi nodes with the highest priority.
 func MembersForQuery() []Node {
 	thisNode := Manager.ThisNode()
 	// If we are running in single mode, just return thisNode
 	if Mode == ModeSingle {
 		return []Node{thisNode}
 	}
 
-	membersMap := make(map[int32][]Node)
+	// store the available nodes for each partition, grouped by
+	// priority
+	membersMap := make(map[int32]*partitionCandidates)
 	if thisNode.IsReady() {
```

Comment:
> having a small warmUp period is still useful as the nodes do a fair amount of work during startup.

note that we only call SetReady or SetReadyIn(warmupPeriod) as the very last thing during startup. in particular, loading the index and starting the input plugins comes first.
So it sounds good, keeping the warmup mechanism, especially for carbon. I'm just skeptical about this particular argument. I think a warmup of 0 should work when using kafka and offset seek back.